src/commands/utils/host.ts

```diff
@@ -3,20 +3,63 @@ import * as vscode from 'vscode';
 const meta = require('../../../package.json');
 
 export async function longRunning<T>(title: string, action: () => Promise<T>): Promise<T> {
-    const options = {
-        location: vscode.ProgressLocation.Notification,
-        title: title
-    };
-    return await vscode.window.withProgress(options, (_) => action());
+  const options = {
+    location: vscode.ProgressLocation.Notification,
+    title: title
+  };
+  return await vscode.window.withProgress(options, (_) => action());
 }
 
 export function getExtensionPath(): string | undefined {
-    const publisherName = `${meta.publisher}.${meta.name}`;
-    const vscodeExtensionPath = vscode.extensions.getExtension(publisherName)?.extensionPath;
+  const publisherName = `${meta.publisher}.${meta.name}`;
+  const vscodeExtensionPath = vscode.extensions.getExtension(publisherName)?.extensionPath;
 
-    if (!vscodeExtensionPath) {
-        vscode.window.showInformationMessage('No Extension path found.');
-        return;
-    }
-    return vscodeExtensionPath;
+  if (!vscodeExtensionPath) {
+    vscode.window.showInformationMessage('No Extension path found.');
+    return;
+  }
+  return vscodeExtensionPath;
 }
+
+export function getSASKey(
+  storageAccount: string,
+  storageKey: string,
+  generateSevenDaySas = false
+) {
+  const storage = require("azure-storage");
+  const startDate = new Date();
+  const expiryDate = new Date();
+
+  startDate.setTime(startDate.getTime() - 5 * 60);
+  let defaultExpirationTime = expiryDate.getTime() + 2 * 24 * 60 * 60 * 1000;
```

Comment:
Because you overwrite this in the 7-day case, you are using the variable as the _actual_ expiration time not just a default; so just call it `expirationTime`.